io-engine/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -681,6 +722,13 @@ impl<'n> Nexus<'n> {
                     })
                 }
             }
+
+            if dev.is_zoned() {
+                //TODO: Implement partitioning zoned block devices. This requires handling drive resources like max active/open zones.
+                warn!("The device '{}' is zoned. Partitioning zoned block devices into smaller devices is not implemented. Using the whole device.", dev.device_name());
+                start_blk = 0;
```

Comment:
This may not work correctly because we have to reserve some space in the child device for metadata. That space calculation is done by `calc_data_partition` and start/end blocks are hence adjusted.